smarts/core/remote_agent_buffer.py

```diff
@@ -87,7 +87,7 @@ def __init__(self, zoo_manager_addrs=None, buffer_size=3):
             conn["channel"], conn["stub"] = get_manager_channel_stub(conn["address"])
 
         self._buffer_size = buffer_size
-        self._replenish_threadpool = futures.ThreadPoolExecutor()
+        self._replenish_threadpool = futures.ThreadPoolExecutor(max_workers=max_workers)
```

Comment:
What I found here was that the `ThreadPoolExecutor` would continue to add threads until it used up all CPUs.

Reply:
I'm wondering if we should use something like `os.cpu_count() / 2` here.  (I don't feel strongly about this either way though.  Hard-coding `4` also seems fine to me.)

Reply:
Yes, I am thinking I might do something like `clamp()` to place CPUs at `os.cpu_count() - 1` between `1` and `4`.

Reply:
Actually this is only for buffering purposes to avoid code slowdowns, hence hard limiting to a max of 4 threads is fine.

Reply:
Yes, on review, I think that it is correct to just roll it back to 4.